rustls/src/enums.rs

```diff
@@ -544,15 +545,40 @@ impl SignatureScheme {
     ///
     /// See RFC8446 s4.2.3.
     pub(crate) fn supported_in_tls13(&self) -> bool {
-        matches!(
-            *self,
-            Self::ECDSA_NISTP521_SHA512
-                | Self::ECDSA_NISTP384_SHA384
-                | Self::ECDSA_NISTP256_SHA256
-                | Self::RSA_PSS_SHA512
-                | Self::RSA_PSS_SHA384
-                | Self::RSA_PSS_SHA256
-                | Self::ED25519
+        let [hash, sign] = self.to_array();
+
+        // This covers both disallowing SHA1:
+        //
+        //  "These values refer solely to signatures which appear in certificates
+        //   (see Section 4.4.2.2) and are not defined for use in signed TLS
+        //   handshake messages"
```

Comment:
This quote doesn't seem to have anything to do with SHA1?

Reply:
Ah, I've cut too much. The full section is:

```
Legacy algorithms:  Indicates algorithms which are being deprecated
      because they use algorithms with known weaknesses, specifically
      SHA-1 which is used in this context with either (1) RSA using
      RSASSA-PKCS1-v1_5 or (2) ECDSA.  These values refer solely to
      signatures which appear in certificates (see Section 4.4.2.2) and
      are not defined for use in signed TLS handshake messages, although
      they MAY appear in "signature_algorithms" and
      "signature_algorithms_cert" for backward compatibility with
      TLS 1.2.
```